platform/mellanox/docker-syncd-mlnx/ecmp_calculator/ecmp_calc.py

```diff
@@ -0,0 +1,492 @@
+#!/usr/bin/env python3
+
+
+from ecmp_calc_sdk import sx_open_sdk_connection, sx_get_active_vrids, sx_router_get_ecmp_id, \
+                          sx_router_operational_ecmp_get, sx_get_router_interface, \
+                          sx_port_vport_base_get, sx_router_neigh_get_mac, sx_fdb_uc_mac_addr_get, \
+                          sx_lag_port_group_get, sx_make_ip_prefix_v4, sx_make_ip_prefix_v6, \
+                          sx_vlan_ports_get, sx_ip_addr_to_str, PORT, VPORT, VLAN, SX_ENTRY_NOT_FOUND
+from port_utils import sx_get_ports_map, is_lag
+from packet_scheme import PACKET_SCHEME
+import json, jsonschema
+import argparse
+import ipaddress
+import re
+import subprocess
+import pprint
+import sys
+
+IP_VERSION_IPV4 = 1
+IP_VERSION_IPV6 = 2
+PORT_CHANNEL_IDX = 1
+VRF_NAME_IDX = 1
+IP_VERSION_MAX_MASK_LEN = {IP_VERSION_IPV4: 32, IP_VERSION_IPV6: 128}
+
+INTF_TABLE = 'INTF_TABLE'
+HASH_CALC_PATH = '/usr/bin/sx_hash_calculator'
+HASH_CALC_INPUT_FILE = "/tmp/hash_calculator_input.json"
+HASH_CALC_OUTPUT_FILE = "/tmp/hash_calculator_output.json"
+
+def check_output_pipe(cmd0, *args):
+    """
+    This function implements check_output API from subprocess module.
+    Input command includes two or more commands connected by shell pipe(s)
+    """
+    popens = [subprocess.Popen(cmd0, stdout=subprocess.PIPE, universal_newlines=True)]
+    i = 0
+    while i < len(args):
+        popens.append(subprocess.Popen(args[i], stdin=popens[i].stdout, stdout=subprocess.PIPE, universal_newlines=True))
+        popens[i].stdout.close()
+        i += 1
+    output = popens[-1].communicate()[0]
+
+    i = 0
+    args_list = [cmd0] + list(args)
+    while popens:
+        current = popens.pop(0)
+        exitcode = current.wait()
+        if exitcode != 0:
+            raise subprocess.CalledProcessError(returncode=exitcode, cmd=args_list[i], output=current.stdout)
+        i += 1
+
+    return output
+
+def is_mac_valid(mac):
+    return bool(re.match("^([0-9A-Fa-f]{2}[:-]){5}([0-9A-Fa-f]{2})$", mac))
+
+def is_ip_valid(address, ip_version):
+    try:
+        if ip_version == IP_VERSION_IPV4:
+            ip = ipaddress.IPv4Address(address)
+            invalid_list = ['0.0.0.0','255.255.255.255']            
```

Comment:
@liorghub how about checking also network and broadcast addresses? (e.g., `network_address` and `broadcast_address` using `ipaddress.ip_network`)

Reply:
ip_network() is a method to get the network address for given ip address.
broadcast_address() is a method to get the  broadcast address for given ip address.
Both broadcast address and network address are valid inputs.